src/ToolBox/ChainProcessor.php

```diff
@@ -33,15 +33,34 @@ public function processOutput(Output $output): mixed
         $response = $output->response;
         $messages = clone $output->messages;
 
+        $options = $output->options;
+
         while ($response->hasToolCalls()) {
             $messages[] = Message::ofAssistant(toolCalls: $response->getToolCalls());
 
             foreach ($response->getToolCalls() as $toolCall) {
+                foreach ($this->toolBox->getMap() as $metadata) {
+                    if ($metadata->name === $toolCall->name
+                        && null !== $metadata->responseFormat
+                    ) {
+                        $options = array_merge($options, [
+                            'response_format' => [
+                                'type' => 'json_schema',
+                                'json_schema' => [
+                                    'schema' => $metadata->responseFormat,
+                                    'name' => $metadata->name,
+                                ],
+                            ],
+                        ]);
+                        break;
+                    }
+                }
+
                 $result = $this->toolBox->execute($toolCall);
                 $messages[] = Message::ofToolCall($toolCall, $result);
             }
 
-            $response = $output->llm->call($messages, $output->options);
+            $response = $output->llm->call($messages, $options);
```

Comment:
this cannot work like this. or maybe it does with the example, but

a) multiple tool calls will interfere with each other
b) the original output format get overwritten

Reply:
just realized why my assumption was wrong. the processor calls the llm not the chain - this is what makes it hard.